vit/readline.py

```diff
@@ -0,0 +1,118 @@
+import string
+import re
+
+class Readline(object):
+    def __init__(self, edit_obj):
+        self.edit_obj = edit_obj
+        word_chars = string.ascii_letters + string.digits + "_"
+        self._word_regex1 = re.compile(
+            "([%s]+)" % "|".join(re.escape(ch) for ch in word_chars)
+        )
+        self._word_regex2 = re.compile(
+            "([^%s]+)" % "|".join(re.escape(ch) for ch in word_chars)
+        )
+
+    def keys(self):
+        return ('ctrl p', 'ctrl n', 'ctrl a', 'ctrl e', 'ctrl b', 'ctrl f',
+                'ctrl h', 'delete', 'ctrl d', 'ctrl t', 'ctrl u', 'ctrl k',
+                'meta b', 'meta f', 'ctrl w', 'meta d')
+
+    def keypress(self, key):
+        if key in ('ctrl p',):
+            text = self.edit_obj.history.previous(self.edit_obj.metadata['history'])
+            if text != False:
+                self.edit_obj.set_edit_text(text)
+            return None
+        elif key in ('ctrl n',):
+            text = self.edit_obj.history.next(self.edit_obj.metadata['history'])
+            if text != False:
+                self.edit_obj.set_edit_text(text)
+            return None
+        elif key in ('ctrl a',):
+            self.edit_obj.set_edit_pos(0)
+            return None
+        elif key in ('ctrl e',):
+            self.edit_obj.set_edit_pos(len(self.edit_obj.get_edit_text()))
+            return None
+        elif key in ('ctrl b',):
+            self.edit_obj.set_edit_pos(self.edit_obj.edit_pos - 1)
+            return None
+        elif key in ('ctrl f',):
+            self.edit_obj.set_edit_pos(self.edit_obj.edit_pos + 1)
+            return None
+        elif key in ('ctrl h',):
+            if self.edit_obj.edit_pos > 0:
+                self.edit_obj.set_edit_pos(self.edit_obj.edit_pos - 1)
+                self.edit_obj.set_edit_text(
+                    self.edit_obj.get_edit_text()[0 : self.edit_obj.edit_pos]
+                    + self.edit_obj.get_edit_text()[self.edit_obj.edit_pos + 1 :])
+            return None
+        elif key in ('delete', 'ctrl d'):
```

Comment:
According to https://readline.kablamo.org/emacs.html ```delete``` and ```ctrl d``` have slightly different functionalities.

Also, I believe the ```delete``` key press is already handled just fine natively by Urwid, so I think we can simply remove that from both here and the ```keys()``` method.

Reply:
Right. Just removed `delete` from readline module.